wgpu-hal/src/gles/command.rs

```diff
@@ -251,14 +251,12 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
     ) where
         T: Iterator<Item = crate::BufferCopy>,
     {
-        //TODO: preserve `src.target` and `dst.target`
-        // at least for the buffers that require it.
         for copy in regions {
             self.cmd_buffer.commands.push(C::CopyBufferToBuffer {
                 src: src.raw,
-                src_target: glow::COPY_READ_BUFFER,
+                src_target: src.target,
```

Comment:
This is problematic. If I create 2 vertex buffers and copy between them, they'll have the same target

Reply:
:thinking: Do you think this will work? Not positive I'm understanding this right.

```rust
       let (src_target, dst_target) = if src.target == dst.target {
            (glow::COPY_READ_BUFFER, glow::COPY_WRITE_BUFFER)
        } else {
            (src.target, dst.target)
        };
        for copy in regions {
            self.cmd_buffer.commands.push(C::CopyBufferToBuffer {
                src: src.raw,
                src_target,
                dst: dst.raw,
                dst_target,
                copy,
            })
        }
```

The examples work on web with that, and I think it would fix your 2 vertex buffer problem, but I'm not sure if it introduces other problems.